ImportExport/ObjectsToPython/ObjectsToPython.FCMacro

```diff
@@ -0,0 +1,344 @@
+                for expression in prop:
+                    addScript("%s.setExpression%s"%(objname, expression))  
+                    
+            elif propname == "_Body":
+                addScript("%s_Group.append(%s)"%(val, objname))
+                          
+            elif (obj.TypeId, propname) == ('PartDesign::Body', 'Group'):
+                addScript("%s_Group = []"%(objname))
+                                                                                     
+            elif objectToText(prop, objectlist) is not None:   
+                if val != defaultval:        
+                    addScript("%s.%s = %s"%(objname, propname, val))   
+                                      
+                                                                                                                            
+def objectToText(obj, objectlist = []):    
+    if obj in objectlist:
+        return obj.Label
+    
+    else:
+        if hasattr(obj, 'Value') and hasattr(obj, 'Unit'):  # how can I test if type(obj) is Base.Quantity ?
+            return "%s"%(obj.Value)
+        else:     
+            if isinstance(obj, str):
+                return "\"%s\""%(obj)
+                
+            if isinstance(obj, bool):
+                if obj:
+                    return "True"
+                else:
+                    return "False"
+            
+            if isinstance(obj, int):
+                return "%d"%(obj)
+    
+            if isinstance(obj, float):
+                return floatstr(obj)                                      
+            
+            if isinstance(obj, FreeCAD.Base.Placement):
+                return "Placement(%s, %s)"%(objectToText(obj.Base), objectToText(obj.Rotation))
+            
+            if isinstance(obj, Part.Point):
+                return "Part.Point(Vector(%s, %s, %s))"%(floatstr(obj.X), floatstr(obj.Y), floatstr(obj.Z))
+            
+            if isinstance(obj, Part.LineSegment):
+                return "Part.LineSegment(%s, %s)"%(obj.StartPoint, obj.EndPoint)
+            
+            if isinstance(obj, Part.Circle):
+                return "Part.Circle(%s, %s, %s)"%(vecstr(obj.Center), obj.Axis, floatstr(obj.Radius))
+            
+            if isinstance(obj, Part.ArcOfCircle):
+                #return "Part.ArcOfCircle(%s, %s, %s)"%(obj.Center, obj.StartPoint, obj.EndPoint)
+                return "Part.ArcOfCircle(%s, %s, %s)"%(objectToText(obj.Circle), obj.FirstParameter, obj.LastParameter)
+            
+            if isinstance(obj, Part.Ellipse):
+                return "Part.Ellipse(%s, %s, %s)"%(vecstr(obj.Center), obj.MajorRadius, obj.MinorRadius)
+            
+            if isinstance(obj, Part.BSplineCurve):
+                poles = ""
+                komma = False
+                for p in obj.getPoles():
+                    if komma:
+                        poles += ", "
+                    poles += vecstr(p)
```

Comment:
Would it be possible to simplify with `', '.join((vecstr(p) for p in obj.getPoles)`?

Reply:
maybe yes, but I prefer my version